### PR TITLE
Feat/sentry with relay

### PR DESCRIPTION
**Pull request by yoan-adfinis**
An alternative to #130 without the nginx requirement. In my experiments, you must disable the relay for the db init to occur, then scale it up. There is probably a nice way to do so via Helm but I ignore it.

I've managed to bypass the CSRF problem but not yet the event.

```
2020-08-04T06:26:48Z [relay_server::actors::project_upstream] DEBUG: updating project states for 1/1 projects (attempt 1)
2020-08-04T06:26:48Z [relay_server::actors::project_upstream] DEBUG: sending request of size 1
2020-08-04T06:26:48Z [relay_server::actors::project] DEBUG: project 2 state updated
2020-08-04T06:26:48Z [relay_server::actors::events] DEBUG: dropped event: could not send event to upstream
  caused by: could not send request to upstream
  caused by: Server disconnected
  caused by: Server disconnected
```

### CODE REVIEW

**Comment by yoan-adfinis** [comment]
so far...

```
2020-08-04T08:06:40Z [actix_net::server::server] INFO: Starting 4 workers
2020-08-04T08:06:40Z [relay_server::actors::project_redis] INFO: redis project cache started
2020-08-04T08:06:40Z [actix_net::server::server] INFO: Starting server on 0.0.0.0:3000
2020-08-04T08:06:40Z [relay_server::actors::controller] INFO: relay server starting
2020-08-04T08:06:40Z [relay_server::actors::connector] INFO: metered connector started
2020-08-04T08:06:40Z [relay_server::actors::events] INFO: event manager started
2020-08-04T08:06:40Z [relay_server::actors::project_local] INFO: project local cache started
2020-08-04T08:06:40Z [relay_server::actors::project_upstream] INFO: project upstream cache started
2020-08-04T08:06:40Z [relay_server::actors::project_cache] INFO: project cache started
2020-08-04T08:06:40Z [relay_server::actors::project_keys] INFO: project cache started
2020-08-04T08:06:40Z [relay_server::actors::relays] INFO: key cache started
2020-08-04T08:06:40Z [relay_server::actors::upstream] DEBUG: got register challenge (token = Z7pd86C-2JHP48u1vqls34iFFxIIqs3B1WlVPLQfRV3ztZZ4KLBuS3fHJ1ApNek-JbpIBkDFU7XJ0pBbqIgPZg)
2020-08-04T08:06:40Z [relay_server::actors::upstream] DEBUG: sending register challenge response
2020-08-04T08:06:40Z [relay_server::actors::upstream] DEBUG: relay successfully registered with upstream
2020-08-04T08:09:07Z [relay_server::actors::project] DEBUG: project 2 initialized without state
2020-08-04T08:09:07Z [relay_server::actors::project] DEBUG: project 2 state requested
2020-08-04T08:09:07Z [relay_server::actors::events] TRACE: queued event
2020-08-04T08:09:07Z [relay_server::actors::project] DEBUG: project 2 state request amended
2020-08-04T08:09:07Z [relay_server::actors::project_upstream] DEBUG: updating project states for 1/1 projects (attempt 1)
2020-08-04T08:09:07Z [relay_server::actors::project_upstream] DEBUG: sending request of size 1
2020-08-04T08:09:07Z [relay_server::actors::project] DEBUG: project 2 state updated
2020-08-04T08:09:07Z [relay_server::actors::events] TRACE: processing json event
2020-08-04T08:09:07Z [relay_server::actors::events] TRACE: sending envelope to kafka
2020-08-04T08:09:07Z [relay_server::actors::store] TRACE: Sending event item of envelope to kafka
```

**Comment by yoan-adfinis** [comment]
I don't see anything... the `snuba-consumer` does nothing as well.

**Comment by yoan-adfinis** [comment]
the `relay` says it registers but nothing shows up there? It's probably expected.

Capture d’écran du 2020-08-06 16-55-24

**Comment by yoan-adfinis** [comment]
@Mokto :tada: I've added a couple of missing services such as ingest-consumer (https://github.com/getsentry/onpremise/pull/421) and transactions-consumer (https://github.com/getsentry/onpremise/pull/595), and make the relay registration wide open (probably unsafe)

Capture d’écran du 2020-08-07 08-47-21

**Comment by Mokto** [comment]
Wow, amazing! Thanks 👍 

Could you add the nginx deployment as an alternative ?
You can just copy my code from the other pull request.

I also think it's time to break the `Values.service` into 4 different ones.

service.sentry
service.relay
service.nginx
service.snuba

What do you think?

File: sentry/values.yaml [hunk 1]
```diff
@@ -52,6 +73,24 @@ sentry:
       maxReplicas: 5
       targetCPUUtilizationPercentage: 50
 
+  ingestConsumer:
+    replicas: 3
```
Comment (Mokto) [inline sentry/values.yaml:77]: Should we put 1 as a default ?

**Comment by yoan-adfinis** [comment]
> Could you add the nginx deployment as an alternative ?

Using nginx when no ingress is configured makes sense for development purposes I guess.

> I also think it's time to break the `Values.service` into 4 different ones.

It totally makes sense.

The next steps could be to integrate the missing consumers first.

**Comment by Mokto** [comment]
I'll close this one ;)

**Pull request closed without merging**